openedx/core/djangoapps/catalog/utils.py

```diff
@@ -210,5 +245,111 @@ def get_run_marketing_url(course_key, user):
     Returns:
         string, the marketing URL, or None if no URL is available.
     """
-    course_run = get_course_run(course_key, user)
-    return course_run.get('marketing_url')
+    course_marketing_urls = get_run_marketing_urls([course_key], user)
+    return course_marketing_urls.get(unicode(course_key))
+
+
+def get_run_marketing_urls(course_keys, user):
+    """
+    Get course run marketing URLs from the course catalog service against course keys.
+
+    Arguments:
+        course_keys ([CourseKey]): A list of Course key object identifying the run whose data we want.
+        user (User): The user to authenticate as when making requests to the catalog service.
+
+    Returns:
+        dict of run marketing URLs against course keys
+    """
+    course_marketing_urls = {}
+    course_catalog_dict = get_course_runs(course_keys, user)
```

Comment:
I agree; I am going to update it to 'catalog_course_runs' omitting the word 'dict' considering best practice not to use such definitive words in variable names.
ref : [CleanCode](https://www.amazon.com/Clean-Code-Handbook-Software-Craftsmanship/dp/0132350882)